File: backend/src/main/java/my/project/models/dto/chineseflashcards/ReviewDTO.java
package my.project.models.dto.chineseflashcards;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.annotation.Nullable;

import java.time.LocalDate;
import java.util.List;

import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;

@JsonInclude(NON_NULL)
public record ReviewDTO(

        @Nullable
        Long id,

        Long userId,

        Integer maxNewWordsPerDay,

        Integer maxReviewWordsPerDay,

        String wordPackName,

        @Nullable
        List<Long> listOfWordId,

        LocalDate dateLastCompleted,

        LocalDate dateGenerated
) {
}

File: library/src/daily-lexika/dtos/flashcards/ReviewDto.ts
import { WordDto } from './WordDto';
import { WordPackUserDto } from './WordPackUserDto';

export interface ReviewDto {
  id?: number;
  userId?: number;
  maxNewWordsPerDay: number;
  maxReviewWordsPerDay: number;
  wordPackId: number;
  wordPackDto: WordPackUserDto;
  listOfWordDto?: WordDto[];
  actualSize: number;
  dateLastCompleted?: string;
  dateGenerated?: string;
}
